addons/mail/static/src/js/client_action.js

```diff
@@ -372,6 +428,52 @@ var ChatAction = Widget.extend(ControlPanelMixin, {
         return $(QWeb.render("mail.chat.Sidebar", options));
     },
 
+    renderTabPanes: function() {
+        var channels = chat_manager.get_channels();
+        function get_channels(type){
+            return _.filter(channels, function(channel){ return channel.type == type; });
+        }
+        var pane_channels = [
+            {type: "dm", channels: get_channels("dm")},
+            {type: "public", channels: get_channels("public")},
+            {type: "private", channels: get_channels("private")}
+        ];
+        var $tabPanes = $(QWeb.render("mail.chat.MobileTabPanes", {
+            channels: pane_channels,
+            moment: moment,
+            widget: this,
+            partner_id: odoo.session_info.partner_id,
+            get_message_body_preview: chat_manager.get_message_body_preview
+        }));
+        this.$(".o_mail_chat_tab_pane").remove();
+        $tabPanes.insertAfter(this.$(".o_mail_chat_content"));
+        if (this.active_mobile_tab == 'channel_inbox' || this.active_mobile_tab == 'channel_starred') {
+            this.switchMobileTab(chat_manager.get_channel(this.active_mobile_tab));
+        } else {
+            this.switchMobileTab(false, this.active_mobile_tab);
+        }
+    },
+
+    // Switching mobile tab and pane for tab
+    switchMobileTab: function(channel, type) {
```

Comment:
missing docstring, and undocumented parameters. also, method should be private, in the Private section. Rename it with a _: _switchMobileTab